api/attestationconfigapi/fetcher.go

```diff
@@ -18,9 +18,23 @@ import (
 
 const cosignPublicKey = constants.CosignPublicKeyReleases
 
+var (
+	// AzureSEVSNP is the Azure SEV-SNP variant.
+	AzureSEVSNP Variant = variant.AzureSEVSNP{}
+	// AWSSEVSNP is the AWS SEV-SNP variant.
+	AWSSEVSNP Variant = variant.AWSSEVSNP{}
+	// GCPSEVSNP is the GCP SEV-SNP variant.
+	GCPSEVSNP Variant = variant.GCPSEVSNP{}
+)
+
+// Variant is a cloud provider specific attestation variant.
+type Variant interface {
+	String() string
+}
```

Comment:
We should drop this and make it an [`fmt.Stringer`](https://pkg.go.dev/fmt#Stringer)

Reply:
I did this in an earlier commit, but it didn't make the the intention so clear. It doesn't suffice to just pass any stringer object.

Reply:
Either we make this a string and accept that it will fail late when running into a 404 on the API call (but have a simple interface without many exports), or we export the proper types that can be used here.